### DISCUSSION TITLE
How to configure Webpack DefinePlugin / vite.define to drop Options API from build

### DISCUSSION THREAD

**Discussion by andystevensname** [nuxt/nuxt#16329] [Questions]
I saw that my rather tiny static site is 1.4MB to download initially, and I'm looking to shed some excess weight. I saw in the Composition API documentation on the Vue website that you can [drop Options API support from a build using a config variable](https://github.com/vuejs/core/tree/main/packages/vue#bundler-build-feature-flags). So, I'm trying to test that out.

Currently I have this in my `nuxt.config`, following the [the Nuxt 3 documentation](https://v3.nuxtjs.org/api/configuration/nuxt-config/#define):

```
vite: {
  define: {
    __VUE_OPTIONS_API__: false,
  }
}
```

I'm unsure if this is actually dropping Options API support in the finished product, as the rendered size doesn't seem to change.

I tried this out with Webpack, just to see if I could see a difference:

```
import { webpack, DefinePlugin } from "webpack";

export default defineNuxtConfig({
  webpack: {
    plugins: [
      new webpack.DefinePlugin({
        __VUE_OPTIONS_API__: false,
      }),
    ],
  }
})
```

The problem with the above is that I can't seem to get the import right, and Nuxt will fail either on find webpack or finding DefinePlugin, so I'm unsure what the best way to import it is.

Either way, does anyone have any idea of how to check to see if this is actually working? I'm wondering if the size saved is going to be that big of a deal to an end user.

**Comment by TheAlexLichter** [comment]
You can run `nuxi analyze`, e.g. via `yarn nuxi analyze` to see the current bundle size of your project (server/client) and which deps are causing each byte 😋

**Reply by andystevensname** [reply]
Thank @manniL. I was able to run the command and realize that I don't think my current vite build settings are doing anything, as far as I can tell.

**Reply by andystevensname** [reply]
@manniL I was able to figure out my major site size issue with `nuxi analyze`, so thank you. I was importing an entire Font Awesome SVG set, rather than just the icons I needed. Fixing this reduced the size of my client bundle by ~1MB, 

I now have two other bits I'd like to optimize for, and that's to prevent the[`NuxtWelcome`](https://nuxt.com/docs/api/components/nuxt-welcome/) component from being included in the bundle, as it's 93.3KB rendered, and reducing my dependence on an SVG animation library. I'm not sure where `NextWelcome` is being used, I don't have a `NuxtWelcome` rendered anywhere on my page.
